### PR TITLE
Modify base64 encoding feature about some input paraments

### PR DESCRIPTION
**Pull request by panpan0721**
1. Add base64 format checking for aad and plaintext paraments
2. Remove all api keys' "_base64" suffix

fix the corresponding test cases
test: passed the unittest

Signed-off-by: panpan0721 <2271409777@qq.com>

### CODE REVIEW

**Review by syan10 — APPROVED** [review]

**Review by yang8621 — CHANGES_REQUESTED** [review]
wait for your more changes

File: ehsm_kms_service/function.js [hunk 1]
```diff
@@ -34,6 +34,7 @@ const base64_encode = (str) => new Buffer.from(str).toString('base64')
 // base64 decode
 const base64_decode = (base64_str) =>
   new Buffer.from(base64_str, 'base64').toString()
+const is_base64 = (base64_str) => /^[a-zA-Z0-9\+\=\/]+$/gi.test(base64_str);
```
Comment (yang8621) [inline ehsm_kms_service/function.js:37]: length is 4X, (X > 0)  
= and == are at the ending

you could search from google.

File: ehsm_kms_service/ehsm_kms_params.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ const keyid = {
   required: true,
 }
```
Comment (yang8621) [inline ehsm_kms_service/ehsm_kms_params.js:23]: I do hope keyid has its own reg check.
we could support it in others patches.

File: ehsm_kms_service/function.js [hunk 3]
```diff
@@ -34,6 +34,15 @@ const base64_encode = (str) => new Buffer.from(str).toString('base64')
 // base64 decode
 const base64_decode = (base64_str) =>
   new Buffer.from(base64_str, 'base64').toString()
+const is_base64 = (base64_str) => {
+  if (base64_str == undefined ||
+      typeof base64_str != 'string' ||
+      base64_str.length == 0 ||
```
Comment (yang8621) [inline ehsm_kms_service/function.js:40]: const aad = {
  type: 'base64',
  maxLength: MAX_LENGTH,
  required: false,
}

aad='' is valid

File: ehsm_kms_service/function.js [hunk 4]
```diff
@@ -333,6 +342,26 @@ const _checkPayload = function (req, res) {
           return false
         }
       }
+      if (currentPayLoad[key].type == 'base64' && payload[key]) {
```
Comment (yang8621) [inline ehsm_kms_service/function.js:345]: why not "else if" in _checkPayload

File: ehsm_kms_service/function.js [hunk 5]
```diff
@@ -333,6 +342,26 @@ const _checkPayload = function (req, res) {
...
+        if (
+          payload[key] != undefined &&
+          ((currentPayLoad[key].maxLength &&
+            payload[key].length > currentPayLoad[key].maxLength) ||
+            (currentPayLoad[key].minLength &&
+              payload[key].length < currentPayLoad[key].minLength))
+        ) {
+          res.send(_result(400, `${key} length error`))
+          return false
+        }
```
Comment (yang8621) [inline ehsm_kms_service/function.js:359]: Can we special handle aad='' here like:

if (!currentPayLoad[key].minLength && payload[key].length === 0)
return true

not sure, please confirm.

**Comment by yang8621** [comment]
one curiosity:
https://github.com/intel/ehsm/blob/2ffb7e4301ae55ba5884ef3a689cfcf76c622d4d/ehsm_kms_service/function.js#L321
what will happen if currentPayLoad is undefined.

File: ehsm_kms_service/function.js (hunk 3 above)
Comment (panpan0721) [reply ehsm_kms_service/function.js:40]: This is a common function for checking valid base64, not only for aad. 
And "required: false," will cover the case that "aad=''" at line 322 - 328

File: ehsm_kms_service/function.js (hunk 4 above)
Comment (panpan0721) [reply ehsm_kms_service/function.js:345]: I just try to follow the code  style like "string", "int" and else other

I dont known why the checking code dont use "else if" before too - -///.

File: ehsm_kms_service/function.js (hunk 5 above)
Comment (panpan0721) [reply ehsm_kms_service/function.js:359]: This case has been covered at line 322 - 328

**Review by yang8621 — APPROVED** [review]

**Pull request merged**